### PR TITLE
Issue 3324: WDA Texts are not taken into account

### PR DESCRIPTION
**Pull request by KDS42**
The WDA texts of the text based entities like labels, buttons, TextView,... have not been included in the ZIP-archive. They are stored as SOTR object and have been added to the github artifacts.

### CODE REVIEW

**Comment by jfilak** [comment]
@larshp Would it be possible to create the repo https://github.com/abapGit-tests/SOTR to move the WDYN tests there?

**Comment by larshp** [comment]
@jfilak I've invited you to the abapGit-tests organization, you should then have access to create repositories there

**Comment by jfilak** [comment]
@larshp  Thanks! I've created the repo https://github.com/abapGit-tests/WDYN because the SOTR class is not an object type handler but a helper class.

@KDS42 We need to move the src/test/zwd_abapgit_test_sotr.wdyn.xml to the new repo. Basically, fork it, create a package $wdyn, clone your fork to your system, move the object to the new package, commit and open a new PR.

**Comment by larshp** [comment]
@jfilak 👍

File: src/objects/zcl_abapgit_sotr_handler.clas.abap [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+  class-methods create_sotr
+    IMPORTING
+      iv_package TYPE devclass
+      it_sotr    TYPE zif_abapgit_definitions=>ty_sotr_tt
+    RAISING
+      zcx_abapgit_exception.
+
+   PROTECTED SECTION.
+     class-methods get_sotr_4_concept
+       importing
```
Comment (jfilak) [inline src/objects/zcl_abapgit_sotr_handler.clas.abap:33]: @larshp can you please explain this check for us?

File: src/objects/zcl_abapgit_sotr_handler.clas.abap [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        concept        = concept
+      IMPORTING
+        header         = ls_header
+      TABLES
+        entries        = lt_entries
+      EXCEPTIONS
+        no_entry_found = 1
+        OTHERS         = 2.
+    IF sy-subrc <> 0.
+      exit.
```
Comment (jfilak) [inline src/objects/zcl_abapgit_sotr_handler.clas.abap:126]: @larshp Actually we should use RETURN, right?

Comment (larshp) [reply src/objects/zcl_abapgit_sotr_handler.clas.abap:126]: Yes, use RETURN instead

see https://help.sap.com/doc/abapdocu_751_index_htm/7.51/en-US/abenleave_processing_blocks.htm

The rule descriptions can be looked up at https://rules.abaplint.org, currently its not possible to add links in the GitHub annotations

File: src/objects/zcl_abapgit_sotr_handler.clas.abap (hunk 1 above)
Comment (larshp) [reply src/objects/zcl_abapgit_sotr_handler.clas.abap:33]: its related to whitespace

Currently there are 5 spaces before "class-methods"
```
     class-methods get_sotr_4_concept
```

indentation in abap is typically done using tabs, which are 2 spaces, ie. all indentation should be at tab positions(spaces MOD 2 = 0)

solution: add another space at start of line

Comment (larshp) [reply src/objects/zcl_abapgit_sotr_handler.clas.abap:33]: similar with 

```   PROTECTED SECTION.```

which have 3 spaces

Comment (larshp) [reply src/objects/zcl_abapgit_sotr_handler.clas.abap:33]: actually just run the SAP standard pretty printer on the source code, with upper case keywords + indentation

File: src/objects/zcl_abapgit_object_wdyn.clas.abap [hunk 3]
```diff
@@ -842,5 +852,14 @@ CLASS zcl_abapgit_object_wdyn IMPLEMENTATION.
     io_xml->add( ig_data = mt_sources
                  iv_name = 'SOURCES' ).
 
+    read table ls_component-comp_metadata-descriptions into ls_description index 1.
+    if sy-subrc = 0.
+      lt_sotr = zcl_abapgit_sotr_handler=>read_sotr_wda( conv #( ls_description-component_name ) ).
```
Comment (jfilak) [inline src/objects/zcl_abapgit_object_wdyn.clas.abap:857]: Well, abaplint claims "lt_sotr" does not exists but in fact "conv" does not exist.

Comment (larshp) [reply src/objects/zcl_abapgit_object_wdyn.clas.abap:857]: CONV does not work on 702, it was introduced in v740sp02

File: src/objects/zcl_abapgit_sotr_handler.clas.testclasses.abap [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  PRIVATE SECTION.
+    METHODS sotr_wda_0001 FOR TESTING.
+    METHODS sotr_cx_0002 FOR TESTING.
+ENDCLASS.
+
+CLASS ltcl_sotr_handler IMPLEMENTATION.
+  METHOD sotr_wda_0001.
+    DATA lt_sotr TYPE zif_abapgit_definitions=>ty_sotr_tt.
+    TRY.
+        lt_sotr = zcl_abapgit_sotr_handler=>read_sotr_wda( iv_object_name = 'ZWD_ABAPGIT_TEST_SOTR' ).
```
Comment (larshp) [inline src/objects/zcl_abapgit_sotr_handler.clas.testclasses.abap:14]: I guess this unit test will fail if run on a system without ZWD_ABAPGIT_TEST_SORT? Which will be almost all systems?

Comment (KDS42) [reply src/objects/zcl_abapgit_sotr_handler.clas.testclasses.abap:14]: I thought the wd component is located in package $WDYN and part of this PR...

Comment (jfilak) [reply src/objects/zcl_abapgit_sotr_handler.clas.testclasses.abap:14]: Oops, I didn't expect Unit Tests relies on external objects.

Comment (KDS42) [reply src/objects/zcl_abapgit_sotr_handler.clas.testclasses.abap:14]: Alternatively we can use an existing WDR-Test-component which exists already in 7.02 - e.g. SALV_WD_TEST_TABLE_SIMPLE. Without the external component we can delete the Unit Test...

Comment (jfilak) [reply src/objects/zcl_abapgit_sotr_handler.clas.testclasses.abap]: @larshp Is it possible to use ABAP OpenSql Test Double Framework? https://help.sap.com/viewer/c238d694b825421f940829321ffa326a/1809.000/en-US/1432ca1fc7b547d493f691cdd09245ae.html

Comment (larshp) [reply src/objects/zcl_abapgit_sotr_handler.clas.testclasses.abap]: no, it was introduced sometime long after 702

suggest just deleting this test case

**Comment by KDS42** [comment]
@Community: can someone check the scenario? E.G. Export your local WD component (including WD application) OFFLINE from system A.  Afterwards importing ARCHIVE into system B, where the component does not exist - are Labels, Texts correctly displayed at Runtime?

**Comment by larshp** [comment]
Hi,
after pull of the example repo I get the following diff, need to sort the elements,

image

Also no SOTR after pull,
image

**Comment by KDS42** [comment]
I tested again - it worked - please see attachment for the steps:
[WDA_SOTR_GITHUB.docx](https://github.com/larshp/abapGit/files/4729843/WDA_SOTR_GITHUB.docx)

**Comment by larshp** [comment]
I think the sorting issue was existing before this PR, so it can be moved to a separate issue

**Comment by larshp** [comment]
anyhow, lets just go for it 🚢

**Comment by larshp** [comment]
and thanks 👍 welcome to abapGit as a contributor

**Pull request merged**